fluent-bit 3.0.3+ fails to build with -Wincompatible-pointer-types and -Wpointer-sign

## Bug Report

**Describe the bug**

When building the [fluent-bit AUR package](https://aur.archlinux.org/packages/fluent-bit) for versions 3.0.3 or 3.0.4 on Archlinux, the build fails with these errors:

```
/build/fluent-bit/src/src/tls/openssl.c:220:40: error: passing argument 1 of ‘SSL_select_next_proto’ from incompatible pointer type [-Wincompatible-pointer-types]
/usr/include/openssl/ssl.h:828:50: note: expected ‘unsigned char **’ but argument is of type ‘const unsigned char **’
```

```
/build/fluent-bit/src/src/tls/openssl.c:222:40: warning: pointer targets in passing argument 3 of ‘SSL_select_next_proto’ differ in signedness [-Wpointer-sign]
/usr/include/openssl/ssl.h:829:55: note: expected ‘const unsigned char *’ but argument is of type ‘char *’
```

This is code introduced in commits 1feb741e8fcf1667b6c7dcc47da314fa7679b601 and 5340b4c1247615637281c14236be0eb9002d1569.

Is the intention to be able to build with -Wincompatible-pointer-types and -Wpointer-sign, or should I relax the build options for the fluent-bit AUR package?

**To Reproduce**
- Clone the fluent-bit AUR package
- Build on Archlinux with `extra-x86_64-build`

**Your Environment**
* Version used: 3.0.3, 3.0.4
* Configuration: -
* Environment name and version (e.g. Kubernetes? What version?): -
* Server type and version: -
* Operating System and version: Archlinux, with openssl 3.3.0-1
* Filters and plugins: -

**Additional context**

N/A

--- a/plugins/in_kubernetes_events/kubernetes_events.c
+++ b/plugins/in_kubernetes_events/kubernetes_events.c
@@ -487,7 +487,7 @@ static int process_events(struct k8s_events *ctx, char *in_data, size_t in_size,
             goto msg_error;
         }
 
-        if (check_event_is_filtered(ctx, item, &ts) == FLB_TRUE) {
+        if (check_event_is_filtered(ctx, item, (time_t *) &ts) == FLB_TRUE) {
             continue;
         }
 

--- a/src/tls/openssl.c
+++ b/src/tls/openssl.c
@@ -217,7 +217,7 @@ static int tls_context_server_alpn_select_callback(SSL *ssl,
     result = SSL_TLSEXT_ERR_NOACK;
 
     if (ctx->alpn != NULL) {
-        result = SSL_select_next_proto(out, 
+        result = SSL_select_next_proto((unsigned char **) out,
                                        outlen,
                                        &ctx->alpn[1], 
                                        (unsigned int) ctx->alpn[0], 

